src/main/java/net/schmizz/sshj/connection/channel/direct/SessionChannel.java

```diff
@@ -93,7 +98,7 @@ public Command exec(String command)
             throws ConnectionException, TransportException {
         checkReuse();
         log.debug("Will request to exec `{}`", command);
-        sendChannelRequest("exec", true, new Buffer.PlainBuffer().putString(command))
+        sendChannelRequest("exec", true, new Buffer.PlainBuffer().putString(command.getBytes(getRemoteCharset())))
```

Comment:
The remote side seems to decode the command using its own charset instead of utf-8. Commands containing filenames with special characters won't work properly using utf-8 on non-utf-8 remote machines.
I'll add the `putString(String s, Charset cs)` method. There will still be a lot of `getRemoteCharset()` calls in SFTPEngine though which could also be replaced by a property.